### DISCUSSION TITLE
translate-x-[value in px] not working in Next JS 13 when using tailwind

### DISCUSSION THREAD

**Discussion by Venkat-Tetali** [tailwindlabs/tailwindcss#12046] [Help]
**What version of Tailwind CSS are you using?**

v3.3.3

**What build tool (or framework if it abstracts the build tool) are you using?**

NextJS 13.4.19, React v18.2.0 

**What version of Node.js are you using?**

v18.17.1

**What browser are you using?**

Chrome

**What operating system are you using?**

MacOS

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**
i tried to use the translate-x-[] syntax by passing in a custom value like 148px to the class and its not working.
exact syntax is transform -translate-x-[148px]

Another fun fact...it works if i hardcode the value directly as above but if I use a template string its not working...for e.g

const translateValue=148;
`
<starting div className=`transform -translate-x-[${translateValue}px]`> Content here <ending div>
`

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

You could consider using the `style` attribute instead like:

```jsx
<div style={{ transform: `translateX(-${translateValue}px)` }}> Content here </div>
```

**Comment by marveal-rab** [comment]
i got the same problem. emmm… but sometimes it works really. Anyone can solve it ? pls…

**Reply by alex-krasikau** [reply]
Before generating the final CSS, Tailwind CSS scans your file to find any appearances of the utilities. In order to add one to the CSS, it needs to see the complete declaration, like `bg-[#000000]`, anywhere in your code. It doesn't work during runtime and is not able to capture `bg-[${black}]`, even if it still generates the correct class name.

Sometimes you can see this working if the resulting class has already appeared in your codebase. Here's the case:

```js
// a.tsx
<div className="... bg-[#000000] ..."></div>

// b.tsx
const black = "#000000";
<div className={`bg-${black} border-${black}`}></div>
```

In this case, Tailwind CSS captures `bg-[#000000]` in the `a.tsx` file and adds it to the generated CSS file. So during runtime, `b.tsx` will be using the same style as the matching class names. However, `border-${black}` will not work because Tailwind CSS didn't generate the corresponding class for this utility.
